packages/core/src/rest/rest-request-builder.ts

```diff
@@ -1,9 +1,24 @@
-import { AxiosRequestConfig } from 'axios';
 import { Destination, DestinationNameAndJwt } from '../scp-cf';
 import { buildAxiosRequestConfig } from '../http-client';
 
-export abstract class RestRequestBuilder {
-  public customHeaders: Record<string, string> = {};
+type FunctionReturnType<ApiT, FnT extends keyof ApiT> = ApiT[FnT] extends (
```

Comment:
Very impressive what is possible with the conditional type. Please add some comment what is done here:
basic open ApiT is the class e.g. `PriceCalculationApi` which has multiple functions e.g. `calculateViaRestWithTenant` represented by the two types `ApiT` and `FnT`. This ensures that on new RestRequestBuilder<PriceCalculationApi>(    PriceCalculationApi,
    'calculateViaRestWithTenant') you need to pass a key as second argument.

Reply:
Request: Could you find or add a second method returning a union type e.g.
AxiosPromise<number|string>. I think the type magic should work but I would like to see a test.

Reply:
Yes, I will add some documentation around this. For the second part I will see what I can do. I guess I can only achieve this by providing a custom service spec file, which I would rather want to do on a separate PR (which can be merged before this one).

Reply:
Yes I think a custom spec file with all facets we want to check makes sense. A seperate PR for this makes also sense.

Reply:
Hey @FrankEssenberger 

I found a couple of problems in the openapi generator (not our additions, but the java thing below). Let's discuss them in the refinement. I documented everything, but would like to refrain from adding tests on this PR, unless you have anything more realistic related to my changes in mind ;) I added a note about you union type idea to the already existing ticket on testing this.